server/src/main/java/org/elasticsearch/search/aggregations/metrics/scripted/ScriptedMetricAggregator.java

```diff
@@ -62,13 +62,26 @@ public boolean needsScores() {
     @Override
     public LeafBucketCollector getLeafCollector(LeafReaderContext ctx,
             final LeafBucketCollector sub) throws IOException {
-        final SearchScript leafMapScript = mapScript.newInstance(ctx);
+        final ScriptedMetricAggContexts.MapScript leafMapScript = mapScript.newInstance(ctx);
         return new LeafBucketCollectorBase(sub, leafMapScript) {
+            private Scorer scorer;
+
+            @Override
+            public void setScorer(Scorer scorer) throws IOException {
+                this.scorer = scorer;
+            }
+
             @Override
             public void collect(int doc, long bucket) throws IOException {
                 assert bucket == 0 : bucket;
+
+                double _score = 0.0;
+                if (scorer != null) {
+                    _score = scorer.score();
```

Comment:
This will cause the score to always be calculated, even when not being used. Instead, in SearchScript, we have the `get_score()` method, which painless converts to a `_score` variable and is only invoked when it is actually used. So, the MapScript should have `setScorer`

Reply:
`get_score()` should work; if it doesn't, it's a bug. @jdconrad Can you check this?

Reply:
Sorry, I was mistaken here.. another quick hack test just now shows variables that start with underscores do work. Not sure what test lead me to the conclusion before that it didn't. That should make this an easy fix. No need to mobilize @jdconrad :)

Reply:
My last commit resolved this.